### PR TITLE
Add command to generate new deprecations

### PR DESCRIPTION
**Pull request by locks**
[edit] I am now ok with shipping this as is.

cc @serenaf

### CODE REVIEW

**Comment by mansona** [comment]
@locks I **love** this 😍 I would be interested in figuring out the "right" way to do these options but also to surface that in the docs for the ember-cli. It could potentially be a good "tracer bullet" for the ember-cli docks improvements

**Comment by locks** [comment]
@mansona I tried a couple approaches and `ember/v3/deprecate-ember-logger` seems like the easiest? 🤔

File: yarn.lock [hunk 1]
```diff
@@ -2314,18 +2314,6 @@ core-js@^2.4.0, core-js@^2.4.1, core-js@^2.5.0:
   version "2.5.3"
   resolved "https://registry.yarnpkg.com/core-js/-/core-js-2.5.3.tgz#8acc38345824f16d8365b7c9b4259168e8ed603e"
 
-core-object@2.0.6:
```
Comment (Turbo87) [inline yarn.lock]: these changes seem unrelated?

File: blueprints/deprecation/index.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    // Return custom template variables here.
+    return {
+      name: extractDeprecationId(options.entity.name)
+    };
+  }
+};
+
+function extractDeprecationId(path) {
+  var pathParts = path.split('/')
+  var name = pathParts[pathParts.length - 1].split('.')[0]
```
Comment (Turbo87) [inline blueprints/deprecation/index.js]: it's not quite obvious what this code extracts. since there are no tests for this it should at least have some comment with an example that explains the behavior.

File: yarn.lock (hunk 1 above)
Comment (locks) [reply yarn.lock]: ✅

File: blueprints/deprecation/index.js (hunk 2 above)
Comment (locks) [reply blueprints/deprecation/index.js]: Can you write the test? 😅
